src/app/hooks/useTasks.ts

```diff
@@ -1,57 +1,84 @@
-import { useState, useEffect } from "react";
 import { Task } from "../types/task";
+import useSWR, { mutate } from "swr";
+import { useState } from "react";
 
-export function useTasks(userId: string | undefined) {
-  const [tasks, setTasks] = useState<Task[]>([]);
-
-  useEffect(() => {
-    if (userId) {
-      fetchTasks();
-    }
-  }, [userId]);
+const fetcher = async (url: string) => {
+  const response = await fetch(url);
+  if (!response.ok) {
+    throw new Error("Failed to fetch tasks");
+  }
+  return response.json();
```

Comment:
**Suggestion: zod parse the response here and return the exact you are expecting**

Reply:
done